atc/exec/check_step.go

```diff
@@ -99,6 +99,10 @@ func (step *CheckStep) run(ctx context.Context, state RunState, delegate CheckDe
 		"step-name": step.plan.Name,
 	})
 
+	if err := delegate.BeforeSelectWorker(logger); err != nil {
```

Comment:
I have concern to pull up `BeforeSelectWorker` to before `delegate.Initializing()`. Looking at

https://github.com/concourse/concourse/blob/fa6ef67835be414278cba484305a8e656ce7a9b3/atc/engine/build_step_delegate.go#L171-L185

`BeforeSelectWorker` means a worker is needed, which implied a real check needs to run. So when only when an in-memory-check really needs to run check, it interacts with DB. If pull up `BeforeSelectWorker`, then all in-memory-check will init db, which will completely eliminate performance improvement of in-memory-check.